tests/test_defineVisits.py

```diff
@@ -24,49 +24,113 @@
+        for i in (347, 348, 349):
+            simple = SerializedDimensionRecord.parse_file(os.path.join(DATADIR, f"exp_{i}.json"))
+            self.records[i] = DimensionRecord.from_simple(simple, registry=self.butler.registry)
+
+    def tearDown(self):
+        if self.root is not None:
+            shutil.rmtree(self.root, ignore_errors=True)
+
+    def assertVisits(self):
+        """Check that the visits were registered as expected."""
+        visits = list(self.butler.registry.queryDimensionRecords("visit"))
+        self.assertEqual(len(visits), 4)
+        self.assertEqual(
+            {visit.id for visit in visits}, {2022040500347, 2022040500348, 2022040500349, 92022040500348}
+        )
+
+        # Ensure that the definitions are correct (ignoring order).
+        defmap = defaultdict(set)
+        definitions = list(self.butler.registry.queryDimensionRecords("visit_definition"))
+        for defn in definitions:
+            defmap[defn.visit].add(defn.exposure)
+
+        self.assertEqual(
+            dict(defmap),
+            {
+                92022040500348: {2022040500348},
+                2022040500347: {2022040500347},
+                2022040500348: {2022040500348, 2022040500349},
+                2022040500349: {2022040500349},
+            },
+        )
+
+    def define_visits(
+        self, exposures: list[DimensionRecord | list[DimensionRecord]], incremental: bool
+    ) -> None:
+        for records in exposures:
+            self.butler.registry.insertDimensionData("exposure", *ensure_iterable(records))
+            # Include all records so far in definition.
+            dataIds = [d for d in self.butler.registry.queryDataIds("exposure", instrument="DummyCam")]
+            self.task.run(dataIds, incremental=incremental)
+
+    def test_defineVisits(self):
+        # Test visit definition with all the records.
+        self.define_visits([[r for r in self.records.values()]], incremental=False)  # list inside a list
+        self.assertVisits()
+
+    def test_incremental_cumulative(self):
+        # Define the visits after each exposure.
+        self.define_visits([exp for exp in self.records.values()], incremental=True)
+        self.assertVisits()
+
+    def test_incremental_cumulative_reverse(self):
+        # In reverse order we should still eventually end up with the right
+        # answer.
+        with self.assertLogs("lsst.defineVisits.groupExposures", level="WARNING") as cm:
+            self.define_visits(list(reversed(self.records.values())), incremental=True)
+        self.assertIn("Skipping the multi-snap definition", "\n".join(cm.output))
+        self.assertVisits()
+
+    def define_visits_incrementally(self, exposure: DimensionRecord) -> None:
+        self.butler.registry.insertDimensionData("exposure", exposure)
+        dataIds = [
+            d
+            for d in self.butler.registry.queryDataIds(
+                "exposure", instrument="DummyCam", exposure=exposure.id
```

Comment:
This should be just the one exposure, right?

Reply:
Yes. I think this is a copy-and-paste from `define_visits` above.